src/code/audio_heap.c

```diff
@@ -1006,40 +1006,42 @@ void AudioHeap_Init(void) {
         reverb->tunedSample.tuning = 1.0f;
         reverb->sample.codec = CODEC_REVERB;
         reverb->sample.medium = MEDIUM_RAM;
-        reverb->sample.size = reverb->windowSize * 2;
+        reverb->sample.size = reverb->windowSize * SAMPLE_SIZE;
         reverb->sample.sampleAddr = (u8*)reverb->leftRingBuf;
         reverb->loop.start = 0;
         reverb->loop.count = 1;
         reverb->loop.end = reverb->windowSize;
 
         if (reverb->downsampleRate != 1) {
             reverb->unk_0E = 0x8000 / reverb->downsampleRate;
-            reverb->unk_30 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, 0x20);
-            reverb->unk_34 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, 0x20);
-            reverb->unk_38 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, 0x20);
-            reverb->unk_3C = AudioHeap_AllocZeroed(&gAudioContext.miscPool, 0x20);
+            reverb->unk_30 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, SAMPLES_PER_FRAME * SAMPLE_SIZE);
+            reverb->unk_34 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, SAMPLES_PER_FRAME * SAMPLE_SIZE);
+            reverb->unk_38 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, SAMPLES_PER_FRAME * SAMPLE_SIZE);
+            reverb->unk_3C = AudioHeap_AllocZeroed(&gAudioContext.miscPool, SAMPLES_PER_FRAME * SAMPLE_SIZE);
```

Comment:
These are all used as `aResample` "state" (`s`), I think the 0x20 size is just its own constant

Sources:

-------------

https://github.com/n64decomp/sm64/blob/1372ae1bb7cbedc03df366393188f4f05dcfc422/include/PR/abi.h#L503-L505
```
 * The state consists of the four following source samples when the algorithm stopped as
 * well as a fractional position
```

---------------

https://github.com/Thar0/oot/blob/b799db384447bf42b2f1b376a362c0e3b4234117/rsp/aspMain.s

This loads 0x20 bytes from the `s` argument (dram address) to `tmpDataPtr`:

```as
/* 04001508 000508 22E10000 */  move    $1, tmpDataPtr
/* 0400150C 00050C 0002B020 */  add     @@dram_addr_save, $zero, @@dram_addr
/* 04001510 000510 2003001F */  li      $3, 0x20 - 1
/* 04001514 000514 001A3C02 */  srl     @@flags, cmd_w0, 0x10
/* 04001518 000518 30EA0001 */  andi    $10, @@flags, A_INIT
/* 0400151C 00051C 1D400005 */  bgtz    $10, @@L04001534        // if (flags & A_INIT) , skip
/* 04001520 000520 00000000 */   nop
/* 04001524 000524 0D0006B5 */  jal     dma_read_start
```

This loads a doubleword (64 bits) aka 4 samples from tmpDataPtr+0:

```as
/* 04001548 000548 CAF01800 */   ldv    $v16[0], (tmpDataPtr)
```

This loads a short (16 bits) from tmpDataPtr+8:

```as
/* 040015A0 0005A0 CAF70F04 */  lsv     $v23[14], 0x8(tmpDataPtr)   // saved pitch_accumulator
```

------------

idk what's going on in details but based on SM64's abi.h docs and Tharo's disassembly and docs, this looks more like just "0x20 bytes of memory" rather than a 16-samples long signal